src/main/java/com/echobox/api/linkedin/connection/versioned/VersionedVideoConnection.java

```diff
@@ -34,13 +57,96 @@ public class VersionedVideoConnection extends VersionedConnection {
   private static final String VIDEOS = "/videos";
   private static final String ACTION_KEY = "action";
   private static final String INITIALIZE_UPLOAD = "initializeUpload";
+  private static final String FINALIZE_UPLOAD = "finalizeUpload";
+  
+  /**
+   * Headers
+   */
+  private static final String HEADER_ETAG = "etag";
   
   /**
    * Instantiates a new connection base.
    *
    * @param linkedinClient the LinkedIn client
    */
-  protected VersionedVideoConnection(VersionedLinkedInClient linkedinClient) {
+  public VersionedVideoConnection(VersionedLinkedInClient linkedinClient) {
     super(linkedinClient);
   }
+  
+  public URN uploadVideo(InitializeUploadRequest initializeUploadRequest, String filePath)
+      throws IOException {
+  
+    Path videoFilePath = Paths.get(filePath);
+    long fileSizeBytes = Files.size(videoFilePath);
+    ValidationUtils.validateVideoFile(fileSizeBytes);
```

Comment:
As the name of this method is validateVideoFile(), I would expect the argument is a video file (could be in type of Path), and then it validates only the size for now. We can more validation in future if it is required.
If the method only validates file size, I would suggest to change it to validateVideoFileSize(), but it is still better to pass the video file as argument.
Personally, I prefer the first one.

Reply:
I do also prefer the first option that you suggested, however, I took the decision to pass in the file size as we need the file size further down in code. Doing it this way saves on an extra IO operation. We can refactor in future when we need to.

Reply:
OK, then may we consider to write as 
`int ValidationUtils.getAndValidateVideoFileSize()`
then return the file size from the method.

Reply:
That doesn't feel very clean to me. It's making the validate method have multiple responsibilities. If it's necessary to change, then I'd prefer to take the hit on an extra IO operation in favour of cleaner code.

Reply:
I can't find anything similar in this SDK, but there are some getAndValidate... methods in main (e.g. MediaItemSuggestion, CreatePropertyRequest, etc.)

Reply:
We should always strive to improve our codebase, so having it in one place is not a good reason to implement it in another.

Reply:
It's true, but from my perspective, it isn't a bad move to return the validated value for a validation method. Might be not very equivalent, but `new URL()` actually does some degree of validation, and it returns a `URL `instance.
Just my 2 cents.

Reply:
The purpose of the method is to validate. Returning the file size is strange to me because it has nothing to do with the act of validating. It's true that at the moment it's only validating the file size, but you could think in future we could be validating on other constraints. 

If the method were validating the following:
- File size
- Video length
- Aspect ratio

Why would returning the file size make sense?

`URL` is doing validation, but the purpose is to ensure that a valid instance can be created.

Reply:
If you're happy with it, I'll pass through the `Path` as you suggested in your first comment.

Reply:
Or you may consider just a simple change 
```suggestion
    ValidationUtils.validateVideoFileSize(fileSizeBytes);
```
You may refactor it when other validation are required.

Reply:
Happy with that suggestion